src/language/concurrency.md

```diff
@@ -144,80 +168,106 @@ void main() [!async!] {
 {% endprettify %}
 
 The `main()` function uses the `await` keyword in front of `_readFileAsync()`
-to let other Dart code (such as event handlers) use the CPU
-while native code (file I/O) executes.
-Using `await` also has the effect of
-converting the `Future<String>` returned by `_readFileAsync()` into a `String`.
-As a result, the `contents` variable has the implicit type `String`.
-
-{{site.alert.note}}
-  The `await` keyword works only in functions that
-  have `async` before the function body.
+to let other Dart code (such as event handlers) use the CPU while native code
+(file I/O) executes. Using `await` also has the effect of converting
+the `Future<String>` returned by `_readFileAsync()` into a `String`. As a
+result, the `contents` variable has the implicit type `String`.
+
+{{site.alert.note}} 
+The `await` keyword works only in functions that have `async` before the 
+function body. 
 {{site.alert.end}}
 
-As the following figure shows,
-the Dart code pauses while `readAsString()` executes non-Dart code,
-in either the Dart virtual machine (VM) or the operating system (OS).
-Once `readAsString()` returns a value, Dart code execution resumes.
+As the following figure shows, the Dart code pauses while `readAsString()`
+executes non-Dart code, in either the Dart virtual machine (VM) or the operating
+system (OS). Once `readAsString()` returns a value, Dart code execution resumes.
 
-![Flowchart-like figure showing app code executing from start to exit, waiting for native I/O in between](/assets/img/language/concurrency/basics-await.png)
+![Flowchart-like figure showing app code executing from start to exit, waiting 
+for native I/O in between](/assets/img/language/concurrency/basics-await.png)
 
-If you'd like to learn more about using `async`, `await`, and futures,
-visit the [asynchronous programming codelab][].
+### Stream syntax
 
-[asynchronous programming codelab]: /codelabs/async-await
+Dart also supports asynchronous code in the form of streams. Streams
+provide values in the future and repeatedly over time. A promise to provide a
+series of `int` values over time has the type `Stream<int>`.
+
+This example will return a new `int` value every second.
+
+<?code-excerpt "lib/stream_syntax.dart"?>
+```dart
+Stream<int> stream = Stream.periodic(const Duration(seconds: 1), (i) => i * i);
+```
+
+#### await-for and yield
 
+Await-for is a type of for loop that executes each subsequent iteration of the
+loop as new values are provided. In other words, it’s used to “loop over”
+streams. In this example, a new value will be returned from the function
+`sumStream` as new values are emitted from the stream that’s provided as an
```

Comment:
Same idea here with the "return" vs "emit" terminology, since `sumStream` doesn't keeping returning a new value, but instead the stream it returns keeps emitting new values.